### PR TITLE
Move _cleanup function to the end of the build

### PR DESCRIPTION
**Pull request by xDaile**

### CODE REVIEW

**Comment by xDaile** [comment]
Add cleanup test for general:error_callback method.

File: iib/workers/tasks/build.py
Comment (yashvardhannanavati) [inline iib/workers/tasks/build.py:1058]: Should we keep calling the _cleanup() function at the start as well? 

Here's what I'm thinking. What if for some reason, the worker shuts down before giving the request time to complete/fail. When it comes back up and a new request is queued, the images from that request that was shut abruptly will still be in the VM. This way, we make sure the working env for a request is always clean

Comment (yashvardhannanavati) [reply iib/workers/tasks/build.py:1058]: @lipoja @zanssa @xDaile thoughts?

Comment (xDaile) [reply iib/workers/tasks/build.py:1058]: In my opinion, it depends on how much we are restricted by the space, for such a (hopefully) rare case. And, also, if we are willing to exchange a little bit of every request performance for a rare chance for space saving. 

Personally, I would keep cleanup just at the end, as I consider this scenario for really rare event, but correct me If I am wrong. 

With new request completed/failed it will be of course deleted anyway.

Comment (yashvardhannanavati) [reply iib/workers/tasks/build.py:1058]: Let's wait for @lipoja and @zanssa 's thoughts before we merge this. :)

Comment (xDaile) [reply iib/workers/tasks/build.py:1058]: I realised that it's may got an reason why cleanup was called before the request. 
Why would anyone in the first place, place cleanup before the request, not after the request? 
Then I realised, that maybe placing it before the request, was probably for some reliability reason.  
So, I am okay also with an option, where cleanup would be called also before the request.

Comment (zanssa) [reply iib/workers/tasks/build.py:1058]: Out of curiosity @yashvardhannanavati, these images are built and tagged with the request id. correct? In the case mentioned above ( a worker was down for some reason), there would be a new built image with a new tag unless I am missing something. Can you explain to me the risk of building the images in that case?

Comment (lipoja) [reply iib/workers/tasks/build.py:1058]: Yes, I agree we should keep the cleanup at the begging as well (as on the end of request)

Comment (zanssa) [reply iib/workers/tasks/build.py]: Yeah, I am okay to have it running before and after the build. Keeping it to run only before the build won't serve the purpose of the function. :)

**Review by lipoja — APPROVED** [review]

**Review by yashvardhannanavati — APPROVED** [review]

**Pull request merged**
